.github/workflows/build.yml

```diff
@@ -1,71 +1,86 @@
-name: build and upload artifacts
+# This is a configuration file for a GitHub Actions workflow.
+# The workflow is triggered when a tag matching the pattern v[0-9]+.* is pushed to the main branch,
+# or when a pull request is opened or when the workflow is manually called.
 on:
   push:
     tags:
-      - v[0-9]+.*
+      - v[0-9]+.* # this regular expression matches tags like v1.0.0, v2.3.4, etc.
     branches:
       - main
   pull_request:
   workflow_call:
 
+# The contents of the repository should be readable by the workflow.
 permissions:
   contents: read
 
+# The workflow consists of three jobs: lint, tests, and build.
 jobs:
-
   lint:
+    # The lint job uses a configuration file in the .github/workflows directory.
     uses: ./.github/workflows/lint.yml
 
   tests:
+    # The tests job uses a configuration file in the .github/workflows directory.
     uses: ./.github/workflows/tests.yml
 
   build:
-    if: ${{ github.ref == 'refs/heads/main' }}
+    # The build job depends on the lint and tests jobs to run successfully before it can start.
```

Comment:
please add this:
    if: ${{ (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/tags/v') }}

We want to build binaries only on main branch and releases

Reply:
ok but we may have some broken CI in main after merge :? how to test that the build is working well in all targets before merge ? @peterjah

Reply:
yes you're right